### PR TITLE
feat: add detailed timeout options

### PR DESCRIPTION
**Pull request by keroxp**
- Added detailed options for all ECS related commands. Previously those were fixed as 300s.

```
--serviceStableTimeout value    max duration seconds for waiting service stable (default: 900) [$CAGE_SERVICE_STABLE_TIMEOUT]
--taskHealthCheckTimeout value  max duration seconds for waiting canary task health check (default: 900) [$CAGE_TASK_HEALTH_CHECK_TIMEOUT]
--taskRunningTimeout value      max duration seconds for waiting canary task running (default: 900) [$CAGE_TASK_RUNNING_TIMEOUT]
--taskStoppedTimeout value      max duration seconds for waiting canary task stopped (default: 900) [$CAGE_TASK_STOPPED_TIMEOUT]
```

- Changed `rollout` cmd to use `deregistration_delay.timeout_seconds` by fetching from ALB target group's attributes if exists (Default: 300s).
- Changed `rollout` cmd not to stop to stoup canary task even if any error occurs during deregistering canary task from target group.

### CODE REVIEW

**Comment by keroxp** [comment]
Diffs from merge are confirmed.

```diff --git a/cage.go b/cage.go
index 1234ab8..48b7e12 100644
--- a/cage.go
+++ b/cage.go
@@ -43,7 +43,7 @@ func NewCage(input *Input) Cage {
        return &cage{
                Input: input,
                Timeout: timeout.NewManager(
-                       10*time.Minute,
+                       15*time.Minute,
                        &timeout.Input{
                                TaskRunningWait:     taskRunningWait,
                                TaskHealthCheckWait: taskHealthCheckWait,
```

**Review by takkyuuplayer — COMMENTED** [review]
I'll check after https://github.com/loilo-inc/canarycage/pull/84 is merged into main.

**Review by alfnets — COMMENTED** [review]
I've made a few comments, but overall it looks good.

File: .vscode/settings.json
[the remark this answers is not in the document]
Comment (keroxp) [reply .vscode/settings.json:1]: No. We need to include settings. Sharing settings is quite common way in modern JS development, as there're essential settings in addition to for each personal environment. Those should be considerable, but useful in most case.

https://github.com/vitest-dev/vitest/tree/main/.vscode
https://github.com/jestjs/jest/tree/main/.vscode

File: cli/cage/commands/flags.go [hunk 1]
```diff
@@ -43,8 +43,52 @@ func CanaryTaskIdleDurationFlag(dest *int) *cli.IntFlag {
...
 }
+
+func TaskRunningWaitFlag(dest *int) *cli.IntFlag {
+	return &cli.IntFlag{
+		Name:        "taskRunningTimeout",
+		EnvVars:     []string{cage.TaskRunningTimeout},
+		Usage:       "max duration seconds for waiting canary task running",
+		Destination: dest,
+		Category:    "ADVANCED",
+		Value:       300,
```
[the remark this answers is not in the document]
Comment (keroxp) [reply cli/cage/commands/flags.go:59]: Right, we should set it as 900 (15 mins).

**Review by alfnets — APPROVED** [review]
LGTM!

File: run_test.go [hunk 2]
```diff
@@ -71,13 +70,13 @@ func TestCage_Run(t *testing.T) {
...
 			Env:  env,
-			ECS:  ecsMock,
+			Ecs:  ecsMock,
 			Time: test.NewFakeTime(),
 		})
 		result, err := cagecli.Run(ctx, &cage.RunInput{
 			Container: &container,
 			Overrides: overrides,
-			MaxWait:   1,
+			// MaxWait:   1,
```
Comment (alfnets) [inline run_test.go:79]: I apologize for bringing this up after approval, but could this possibly be something that was forgotten to be deleted?

**Comment by keroxp** [comment]
@takkyuuplayer @alfnets A comment out was omitted. Could we merge?

**Review by takkyuuplayer — APPROVED** [review]
LGTM!

**Pull request merged**